library/components/TTable.vue

```diff
@@ -0,0 +1,1018 @@
+<template>
+  <div>
+    <!-- The TColumnConfig that ends up in the slot is an unrecognized component, the back end must be adding them based on what is used in the page
+    so add this here just to tell the back end that this component uses the TColumnConfig component -->
+    <TColumnConfig :columnConfig="['test']"></TColumnConfig>
+    <slot name="columnConfig" :setColumnConfig="setColumnConfig"></slot>
+
+    <TSearch
+      v-if="can_search"
+      ref="easySearch"
+      @updateSearchTerm="updateSearchTerm"
+      :highlight-query-selector="
+        enableSearchHighlight ? '#tableContainer' : null
+      "
+      :highlight-options="highlightOptions"
+    ></TSearch>
+
+    <div
+      v-if="isDataAvailable"
+      id="tableContainer"
+      ref="tableContainer"
+      :style="columnWidthsStyle"
+    >
+      <!-- Title -->
+      <div v-if="title !== null" id="title" class="spanAllColumns center_cell">
+        {{ title }}
+      </div>
+
+      <!-- Empty div to keep the headers lined up with their columns when there are exapnd/collapse buttons  -->
+      <div v-if="can_collapse_detail_rows"></div>
+      <!-- Empty div to keep the headers lined up with their columns when there are radio buttons  -->
+      <div v-if="show_radio_buttons"></div>
+      <!-- Toggle all of the check boxes  -->
+      <input
+        v-if="show_checkboxes"
+        ref="checkAll"
+        v-model="allChecked"
+        type="checkbox"
+        :indeterminate.prop="someChecked"
+      />
+      <!-- Headings -->
+      <div
+        v-for="(column, index) in internalColumns"
+        :key="column.header"
+        :ref="'headerCell_' + index"
+        class="headerCell"
+        :class="generateHeaderClasses(column.property, index)"
+      >
+        <slot :name="generateSlotName('header', column.header)" v-bind="column">
+          {{ column.header }}
+        </slot>
+
+        <span
+          v-if="column.sort || column.initialSort"
+          @click="reverseSort(column.property)"
+        >
+          <slot
+            v-if="arrowDirection[column.property] === 'asc'"
+            name="sortAscendingIcon"
+            v-bind="column"
+          >
+            <div class="icon move_up">&#8964;</div>
+          </slot>
+          <slot v-else name="sortDescendingIcon" v-bind="column">
+            <div class="icon move_down">&#8963;</div>
+          </slot>
+        </span>
+      </div>
+
+      <!-- No table data -->
+      <div
+        v-if="
+          !internalRows ||
+          internalRows.length === 0 ||
+          !displayRows ||
+          displayRows.length === 0
+        "
+        class="spanAllColumns center_cell"
+      >
+        <div><i class="fa-solid fa-inbox"></i></div>
```

Comment:
```suggestion
        <div><i class="i-fa-solid i-fa-inbox"></i></div>
```

Should be like this in the EM (using antfu's pure CSS icons).

Reply:
👍